examples/reduction/reduction/reduction.py

```diff
@@ -0,0 +1,268 @@
+# Copyright 2023 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from enum import IntEnum
+from typing import Any
+
+import cunumeric as num
+import pyarrow as pa
+
+import legate.core.types as ty
+from legate.core import Array, Store
+
+from .library import user_context as context, user_lib
+
+
+class OpCode(IntEnum):
+    BINCOUNT = user_lib.cffi.BINCOUNT
+    CATEGORIZE = user_lib.cffi.CATEGORIZE
+    HISTOGRAM = user_lib.cffi.HISTOGRAM
+    MATMUL = user_lib.cffi.MATMUL
+    MUL = user_lib.cffi.MUL
+    SUM_OVER_AXIS = user_lib.cffi.SUM_OVER_AXIS
+
+
+class _Wrapper:
+    def __init__(self, store: Store) -> None:
+        self._store = store
+
+    @property
+    def __legate_data_interface__(self) -> dict[str, Any]:
+        """
+        Constructs a Legate data interface object from a store wrapped in this
+        object
+        """
+        dtype = self._store.type.type
+        array = Array(dtype, [None, self._store])
+
+        # Create a field metadata to populate the data field
+        field = pa.field("Array", dtype, nullable=False)
+
+        return {
+            "version": 1,
+            "data": {field: array},
+        }
+
+
+def to_cunumeric_array(store: Store) -> num.ndarray:
+    return num.asarray(_Wrapper(store))
+
+
+def print_store(store: Store) -> None:
+    print(to_cunumeric_array(store))
+
+
+def _sanitize_axis(axis: int, ndim: int) -> int:
+    sanitized = axis
+    if sanitized < 0:
+        sanitized += ndim
+    if sanitized < 0 or sanitized >= ndim:
+        raise ValueError(f"Invalid axis {axis} for a {ndim}-D store")
+    return sanitized
+
+
+def sum_over_axis(input: Store, axis: int) -> Store:
+    """
+    Sum values along the chosen axis
+
+    Parameters
+    ----------
+    input : Store
+        Input to sum
+    axis : int
+        Axis along which the summation should be done
+
+    Returns
+    -------
+    Store
+        Summation result
+    """
+    sanitized = _sanitize_axis(axis, input.ndim)
+
+    # Compute the output shape by removing the axis being summed over
+    res_shape = tuple(
+        ext for dim, ext in enumerate(input.shape) if dim != sanitized
+    )
+    result = context.create_store(input.type.type, res_shape)
+    to_cunumeric_array(result).fill(0)
```

Comment:
same comment here. It seems more intuitive to me to create a cunumeric zeros and then get the store from it.